### PR TITLE
Announcement on lavaland and off station content.

### PR DESCRIPTION
**Pull request by optimumtact**
Okay so this has been something spinning around in my head for a while.

Basically, I think we're in a situation where people are spending a lot of time focusing on stuff that isn't on the station and doesn't interact with our core gameplay. We have lost a lot of focus, and it's hampering our momentum in terms of actually changing and improving things about the station, where most of the playerbase spends most of their time.

So as such, I am asking all maintainers to deny things that are not laser focused on the station and crew.

We will not carry code or content outside of the station unless it's burden is very small and minimal, that means lavaland ruins, space ruins and away missions.

You will notice I have already killed off lavaland dwarves and icemoon PR's, both are endemic of the focus on stuff that is outside the station.

Note: This does not mean these ideas are dead forever or the content they have is not useable.

space has already expressed interest in taking a station and putting it *on* the icemoon. This is totally acceptable, because the content is now focused in on the the station and the crew and I think we are past overdue such an idea.

Mixing the content in with the station makes carrying all the sprites and code worth it, because it's much more approachable.

anyway, that's it, feel free to complain here I guess, please don’t' be rude.

### CODE REVIEW

**Comment by bluetshirtguy** [comment]
Would you still allow tweaks to existing content? what about adding onto existing content?

**Comment by PKPenguin321** [comment]
👍 to this, though it is a bit sad that Kor's vision for planetstation won't come to fruition with this. Not like it ever would have done that in the first place without a design lead like him actively working on it, but now it's even deader, so yeah

**Comment by Couls** [comment]
> 👍 to this, though it is a bit sad that Kor's vision for planetstation won't come to fruition with this. Not like it ever would have done that in the first place without a design lead like him actively working on it, but now it's even deader, so yeah

Actually I think planetstation has a higher chance to happen with this, like oranges said moving the station to lavaland/icemoon would be an acceptable move because it still focuses on the station itself. The way we were going lavaland was being used as a content dump that had no place in the core game or theme or the game and planetstation would've never happened if we kept filling lavaland with the equivalent of a feature landfill

**Comment by optimumtact** [comment]
tweaks are fine, I don't want to see things that are massive new content dumps like dwarves or icemoon.

**Comment by wesoda25** [comment]
damn I picked a terrible time to work on a ruin, oh well

**Comment by Farquaar** [comment]
I'm totally down for a station on the Icemoon. We have all the department winter coats already made, and if the moon is cold enough, then we may just see temperature become a threat to replace depressurization.

**Comment by Ziiro** [comment]
I understand the reasoning but I don't really agree with it. I think the concept of a living _server_ with multiple things happening in parallel with multiple locations is far more interesting than "Everything happens on the station all the time". I understand this isn't removing it, but limiting its development leads us down this path.

**Comment by TheVekter** [comment]
I'm cool with this. Lavaland (and by extension mining) is in a really weird place right now and I don't think we need more content until we fix the endemic issues with the job itself.

**Comment by ATH1909** [comment]
Will this apply to PRs that are already open, like zxaber's ash walker rework PR that's been open (and has been actively worked on) for over two weeks and is pretty much ready to be merged?

**Comment by ATH1909** [comment]
~~Also, imagine ordering the removal of cloning, then announcing a ghost role content freeze.~~

**Comment by tmtmtl30** [comment]
i don't understand the ultimatum on the lava dwarves (i know that's a different PR, but since i'm talking about your choices related to this, and not those of the PR author, i'm saying this here). stopping development on off-station content? sure, i get that regardless of whether or not i fully agree with it. if you want to incentivize more on-station content, then this seems like a good way to fulfill that desire. but saying "you have to remove the ashwalkers to add the lava dwarves" doesn't make sense to me. all the development time has already been invested, and ashwalkers have existed for long enough that cutting them out wouldn't really lighten the workload of bugfixers much. they've already been balanced, they're a fully implemented feature already in the game. features like lava dwarves / ashwalkers don't need to be zero-sum, and i don't see any good reason to make them that way. it's retroactively punitive. again, not talking about the off-station freeze, that's its own thing, but specifically the ultimatum given on the lava dwarves PR.

**Comment by optimumtact** [comment]
development time is a small portion of the actual cost of a feature, which is actually paid over the entire life of a feature as the maintenance cost.

Which is why it is a zero sum game.

**Comment by optimumtact** [comment]
Let me use an analogy to explain.

Lets consider each "feature" to be a ping pong ball, on a ping pong table.

each table has a group of developers assigned, and we'll assume the work of keeping a pingball bouncing across that table is roughly equivalent to the work of maintaining and fixing code.

Every feature that gets developed, sure, it takes a while to make the bat and the ball, but the most time that is spent? Keeping that ball bouncing back and forth across the table, for however many years the item is in the game (which we have been developing continuously since the late 2000's).

The main pingpong arena, with the most spectators, is the station, which is obviously where I want most of my developers keeping as many ping pong balls as possible bouncing across that table, to keep the spectators interested.

The secondary table, in the smaller arena, with less people watching, is lavaland, sure I want to have one or two ping pong balls on that table, but it needs to be commensurate with he number of people who actually watch ping pong lavaland.

As to your comment:
Changing one bat and ball combo out for another, that's a different brand of bat, that doesn't "increase" the amount of effort spent bouncing balls across the table. Now sure, not every ball and bat is identical, but these two are pretty similar in scope.

Adding the second ball in? that's another pair of bats and people to bounce them. The task on the maintainers is now increased.

All i'm saying in this announcement, is no more ping pong balls being added to the smaller table, lets focus on more ping pong balls on the main table, and what you can do is swap out brands on the smaller table.

anyway i've beaten the analogy to death so goodnight.

**Comment by Ty-the-Smonk** [comment]
Mad man finally did it

**Comment by FantasticFwoosh** [comment]
That was a long winded way of saying you want to shorten the space between your metaphyiscal ping pong tables to join together by playing doubles and compromising station space & lavaland space on the same plane. 

Its safe to say i dont think most people would think its a bad idea in itself, but you have too much confounding bias to hand-wave away arguements people come up to you with to actually ever succeed the ambitious project and recieve the shared credit for completing your goals.

**Comment by OverlyGeneric** [comment]
I don't think this would change much in the long run apart from slowing down content.

People will just add new drinks, food, switch up a map design, and the occasional attempt to add something new to xenobiology.

The only difference with mining is it's rather spoiled with options when the round extends beyond an hour.

That being said, you could show us what to do instead of telling us. It would be a nice change of pace to know what features you want.

**Comment by ExcessiveUseOfCobblestone** [comment]
For the best tbh, I'm personally not even interested in major station content that's bound to events or gamemodes. I would like core gameplay fleshed out, preferably in ways that challenge the player and have situational routes instead of direct upgrades (IE perhaps you need to focus on product thoroughput vs high quality products).

I know not everyone likes this but it would be better if you CAN'T get everything within a round, and each round may provide different challenges that should encourage progressing different ways.

**Comment by anconfuzedrock** [comment]
Content in tf2 is the dragon's fury, a flamethrower which rewards the pyro for good aim, which therefore rewards the opponent for dodging as the pyro's aim is based on how evasive their opponent is. It also reduces the ability of the pyro to reflect projectiles, meaning an enemy projectile user who takes into account that the pyro has the dragon's fury can gain an advantage. 

This is station content (if station content was GOOD)

Note how the dragon's fury does not reward the pyro for playing a tetris minigame in the respawn lobby for 60% of the match that has no player input from anyone else on the server and results in the pyro getting infinite critical hits.

This is the ash drake.

**Comment by OverlyGeneric** [comment]
> I don't think that lavaland and off station content are the only thing with this issue- xenobiology and virology are also isolated in a similar vein.

There are valid reasons for having them isolated like the various dangerous things like the xenomorph egg, a virus breakout, and lavaland megafauna.

I am willing to assume no one wants to compromise of making any of these safer just for more interaction despite them being isolated. 

It's even a worse idea for forcing interaction as that just makes it tedious and disengaging. Mining is great because of its options like harvesting ores, exploring ruins, hunting fauna, atmos harvesting, and outpost building.

The best solution is to give more paths to choose for roles to contribute to the station without sacrificing the original purpose of it.

**Comment by ExcessiveUseOfCobblestone** [comment]
box has viro and xeno not in their private playpens and I don't think it has a significant difference in virus/slime/xenomob outbreaks.

The problem is these three can get all their loot within their private playpens barring viro who has to go out literally once for their uranium/silver run, and use the fact they are in private playpens to justify the need to 'qol' their role to the point they don't need any assistance from others.

**Comment by OverlyGeneric** [comment]
> The problem is these three can get all their loot within their private playpens barring viro who has to go out literally once for their uranium/silver run, and use the fact they are in private playpens to justify the need to 'qol' their role to the point they don't need any assistance from others.

Xenobiology and virology, maybe. I've haven't seen any mining QOL prs recently.

you can call me blind if you want. i still think more options would be a good choice. #49581 is a good step toward.

**Comment by optimumtact** [comment]
Mining QoL is not banned, I'm trying to stop people stabbing the corpse more, not patching the existing bleeding.

**Comment by anconfuzedrock** [comment]
It's not about how physically isolated viro and xeno are entirely, it's how they have nothing to do with the rest of the crew whatsoever. At most viro asks chemists for the one thing they want which is robotic and the interaciton is just a few seconds of time wasted and a bit of unreliability in getting the stuff as viro

**Comment by BadSS13Player** [comment]
Is this a permanent ban or a content freeze?

**Comment by ExcessiveUseOfCobblestone** [comment]
I can't speak for headcoders but my opinion is that even if it isn't permanent, you should not be making large-scale features that aren't meant to be supported by the primary focus of the game (right now the primary is the station). Onstation crew should be the main focus of the game unless your PR explicitly changes that focus (IE bringing iceland to the station).

I certainly don't want another lavaland incident where we have to make magic economies and weird scaling just to easily support it when it starts trying to interact with the rest of the world.

**Comment by BadSS13Player** [comment]
I don't get this ban. Any role in one round won't (or straight up can't) enjoy most content in the game. Why would off-station content be any different from some niche on-station content?

**Comment by optimumtact** [comment]
did you not read my pingpong analogy

**Comment by BadSS13Player** [comment]
I did. I do not think it is a fair analogy, since not everyone is playing the entire station at once. Only MDs are going to heal/revive people, only engineering is going to do constructiion or sm/atmos autism and so on, just like only shaft miners are going to go to lavaland and only the assistants/scientists who go exploring are going to find space ruins.

**Comment by optimumtact** [comment]
please read it again

**Comment by BadSS13Player** [comment]
I read it again and it didn't become any clearer to me. How I see it is that there isn't a big station ping pong table but multiple tables that are medical, engineering, etc.

**Comment by optimumtact** [comment]
okay, but that's not how I see it.

**Comment by anconfuzedrock** [comment]
well if every department is a ping pong table then mining is a table in a different fucking building

**Comment by EgorDinamit** [comment]
I guess the problem is that people aren't watching those "tables" because there is nothing to watch, adding more content won't be a bad decision to make.
Of course, that would probably leave the station uninhabited, but still, we have situations like green shift and lavaland or space in such cases can keep players entertained.

**Comment by cacogen** [comment]
👶🏻🏓🍊🏓👶🏻

**Comment by ATH1909** [comment]
redacted

Now that I'm ungitbanned, I feel like I can express this point here.

Aside from zxaber's new PR, what new content have ash walkers been getting that dwarves wouldn't also be getting (access to)? Ash walkers have been around for ages, and so their content has mostly settled (aside from some "Tribal" additions **that dwarves have access to**). Search for "ash walker" (or "ashwalker") in the pull requests section, and see how many PRs that are specifically ash walker-focused have been made since mid-2019.

Dwarves, on the other hand, are gonna be buggy (as every big addition to the game tends to be shortly after being merged) and are probably need/inspire the creation of (dwarves are deeply entrenched in pop culture, of course they're gonna inspire some references and extra features) some extra content in future PRs. If you think that coders will be unable to restrain themselves from making ash walkers vs. dwarves content or something, why not just code it so that dwarves and ash lizards can't both spawn on lavaland (and maybe revert that change/remove that restriction after the ghost role freeze ends) in the same round?

In short, you cite this announcement as a reason to not keep both ash lizards and dwarves around, yet replacing ash lizards with dwarves seems to be something that would conflict with your stated purpose(s) for making this edict.

**Comment by optimumtact** [comment]
you clearly didn't understand the analogy at all.

**Comment by ATH1909** [comment]
> you clearly didn't understand the analogy at all.

Could you, er, elaborate on _where_ I'm failing to understand the analogy? Just saying that I don't understand your anology doesn't help me actually understand your analogy or make your reasoning for this announcement any more clear.

**Comment by BadSS13Player** [comment]
I think this whole thing is a troll on the same level as #49500.

**Comment by trollbreeder** [comment]
Okay, here is my take on the ping pong analogy.

There is a ping pong ball competition (a round) in a building (the game) for around an hour and a half. On the first floor (the station) there are a few ping pong ball tables (content), most if not all being used. 
On the second floor (lavaland) there are a few more ping pong ball tables (lavaland content). Those who lost the competition on the first floor (dying) can participate in the ping pong tables on the second floor (ghost roles) or spectate the main floor and second floor events (remaining a ghost).

One day, the building decides to replace a popular ping pong table with a different ping pong table. The second floor players notice, and they find themselves disliking this new ping pong table. The new ping pong table is more prone to breaking sometimes (bugs and issues) than the old ping pong table. This makes the second floor players play on that table less. This also makes some of the second floor players angry, so they complain to the competition manager (headcoder), the table mechanics (maintainers), and/or the ping pong table manufacturers (coders) about the new ping pong table.

Either the competition managers keep the new table and try to keep it from breaking (maintaining content) or they go back on their word and return the old table (reverting PRs), or they can bring back the old table and keep the new table (which is more content overall, which is more interesting but in theory requires more maintenance).

Should we make the second floor event more interesting to watch for spectators by adding more content? Should we abolish the second floor event? Etc, etc, etc.

**Comment by Jalleo** [comment]
What this PR means and says is this: The station itself needs either refreshed features or new ones to replace the stagnant ones. Which is correct and true. Prove me wrong.

What I would like to see is references to talked about features that was proposed to point coders to what could be done to increase engagement which increases interactivity.

The analogy of the pyro from TF2 is that we need features which rewards direct engagement between players in a method. The newer monetary system can do that, so does the bounty system. 

What mechanic can you work out is wholly seperate in how it functions that could be given side paths of seperate divergent development in rounds or be reworked to mean that it can be improved by multiple sections working well gives it a cumulative positive to such a mechanic?

That is what can be done.

**Comment by Ty-the-Smonk** [comment]
I think it’s clear that he wants people to solely focus on station content and not off station stuff that only 2 consistent static’s will ever do or use
